internal/v2/s2av2.go

```diff
@@ -194,12 +200,19 @@ func (c *s2av2TransportCreds) ServerHandshake(rawConn net.Conn) (net.Conn, crede
 	}
 	ctx, cancel := context.WithTimeout(context.Background(), *S2ATimeout)
 	defer cancel()
-	cstream, err := createStream(ctx, c.s2av2Address)
+
+	var s2AStream stream.S2AStream
+	var err error
+	if c.getS2AStream == nil {
+		s2AStream, err = createStream(ctx, c.s2av2Address)
+	} else {
+		s2AStream, err = c.getS2AStream(ctx, c.s2av2Address)
+	}
```

Comment:
nit: maybe factor out these lines of code, since it's the same between client and server handshake

Reply:
Done.

I integrated this part of logic into `createStream` function.